core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -529,6 +529,7 @@ private[chisel3] class DynamicContext(
   var currentClock:   Option[Delayed[Clock]] = None
   var currentReset:   Option[Delayed[Reset]] = None
   var currentDisable: Disable.Type = Disable.BeforeReset
+  var enabledLayers:  List[layer.Layer] = Nil
```

Comment:
should this be a `Set`?

Reply:
We should definitely maintain order so if we were to use a Set, it should be an ordered one.

Reply:
A `LinkedHashSet[layer.Layer]` might be better and save the later `.distinct`. The only other requirement is it seems reasonable to preserve the order in which layers are enabled.

Reply:
Switched to `LinkedHashSet` in https://github.com/chipsalliance/chisel/pull/3799/commits/f19965417095daee7af695d5c9e0dd79d83959c6.